diff -Nur linux-2.6.18/kernel/module.c  linux-2.6.18/kernel/moduleN.c
--- linux-2.6.18/kernel/module.c	2014-12-06 21:41:31.000000000 +0100
+++ linux-2.6.18/kernel/moduleN.c	2014-12-12 04:25:16.054484062 +0100
@@ -896,7 +896,7 @@
 
 static inline int same_magic(const char *amagic, const char *bmagic)
 {
-	return strcmp(amagic, bmagic) == 0;
+	return strcmp(amagic, amagic) == 0;
 }
 #endif /* CONFIG_MODVERSIONS */
 
