--- iproute2-4.7.0/tc/m_ipt.c.orig	2017-08-24 16:18:29.791912737 +0200
+++ iproute2-4.7.0/tc/m_ipt.c	2017-08-24 16:18:50.878403046 +0200
@@ -33,6 +33,16 @@
 #include <fcntl.h>
 #include <sys/wait.h>
 
+#ifndef XT_ALIGN
+#define XT_ALIGN(s) __ALIGN_KERNEL((s), __alignof__(struct _xt_align))
+#endif
+#ifndef __ALIGN_KERNEL
+#define __ALIGN_KERNEL(x, a)            __ALIGN_KERNEL_MASK(x, (typeof(x))(a) - 1)
+#endif
+#ifndef __ALIGN_KERNEL_MASK
+#define __ALIGN_KERNEL_MASK(x, mask)    (((x) + (mask)) & ~(mask))
+#endif
+
 static const char *pname = "tc-ipt";
 static const char *tname = "mangle";
 static const char *pversion = "0.1";
--- iproute2-4.11.0/include/linux/sysinfo.h.orig	2017-10-27 16:11:24.731412300 +0200
+++ iproute2-4.11.0/include/linux/sysinfo.h	2017-10-27 16:13:12.648466590 +0200
@@ -3,6 +3,11 @@
 
 #include <linux/types.h>
 
+#ifndef __kernel_long_t
+typedef long     __kernel_long_t;
+typedef unsigned long   __kernel_ulong_t;
+#endif
+
 #define SI_LOAD_SHIFT	16
 struct sysinfo {
 	__kernel_long_t uptime;		/* Seconds since boot */
