--- linux-2.6.18/lib/string.c.orig	2017-08-24 09:46:26.307300965 +0200
+++ linux-2.6.18/lib/string.c	2017-08-24 09:48:05.889802665 +0200
@@ -60,6 +60,20 @@
 EXPORT_SYMBOL(strnicmp);
 #endif
 
+#ifndef __HAVE_ARCH_STRCASECMP
+int strcasecmp(const char *s1, const char *s2)
+{
+	int c1, c2;
+
+	do {
+		c1 = tolower(*s1++);
+		c2 = tolower(*s2++);
+	} while (c1 == c2 && c1 != 0);
+	return c1 - c2;
+}
+EXPORT_SYMBOL(strcasecmp);
+#endif
+
 #ifndef __HAVE_ARCH_STRCPY
 /**
  * strcpy - Copy a %NUL terminated string
--- linux-2.6.18/include/linux/string.h.orig	2017-08-24 09:43:27.717780589 +0200
+++ linux-2.6.18/include/linux/string.h	2017-08-24 09:45:45.906314369 +0200
@@ -47,6 +47,9 @@
 #ifndef __HAVE_ARCH_STRNICMP
 extern int strnicmp(const char *, const char *, __kernel_size_t);
 #endif
+#ifndef __HAVE_ARCH_STRCASECMP
+extern int strcasecmp(const char *s1, const char *s2);
+#endif
 #ifndef __HAVE_ARCH_STRCHR
 extern char * strchr(const char *,int);
 #endif
