diff --git a/include/bcm_ioctls.h b/include/bcm_ioctls.h
index 81b0dd0..d274d04 100755
--- a/include/bcm_ioctls.h
+++ b/include/bcm_ioctls.h
@@ -44,7 +44,11 @@ typedef enum {
 	STREAMTYPE_MPEG4_Part2 = 4,
 	STREAMTYPE_VC1_SM = 5,
 	STREAMTYPE_MPEG1 = 6,
+#ifdef DREAMBOX
+	STREAMTYPE_MPEG4_H265 = 22,
+#else
 	STREAMTYPE_MPEG4_H265 = 7,
+#endif
 	STREAMTYPE_VB8 = 8,
 	STREAMTYPE_VB9 = 9,
 	STREAMTYPE_XVID = 10,
@@ -70,7 +74,11 @@ typedef enum {
 	AUDIOTYPE_DTS_HD = 0x10,
 	AUDIOTYPE_WMA = 0x20,
 	AUDIOTYPE_WMA_PRO = 0x21,
+#ifdef DREAMBOX
 	AUDIOTYPE_AC3_PLUS = 0x22,
+#else
+	AUDIOTYPE_AC3_PLUS = 7,
+#endif
 	AUDIOTYPE_AMR = 0x23,
 	AUDIOTYPE_RAW = 0xf
 } audio_stream_type_t;
