--- strace-4.18/netlink.c.orig	2017-10-27 11:41:26.478162297 +0200
+++ strace-4.18/netlink.c	2017-10-27 11:43:15.739127954 +0200
@@ -29,6 +29,7 @@
 
 #include "defs.h"
 #include "netlink.h"
+#include <linux/types.h>
 #include <linux/audit.h>
 #include <linux/rtnetlink.h>
 #include <linux/xfrm.h>
--- strace-4.18/tests/netlink_audit.c.orig	2017-10-27 11:41:33.877956797 +0200
+++ strace-4.18/tests/netlink_audit.c	2017-10-27 11:43:35.137590043 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <sys/socket.h>
+#include <linux/types.h>
 #include <linux/audit.h>
 #include "netlink.h"
 
--- strace-4.18/tests/ptrace.c.orig	2017-10-27 11:41:45.512633681 +0200
+++ strace-4.18/tests/ptrace.c	2017-10-27 11:43:35.139589987 +0200
@@ -38,6 +38,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include "ptrace.h"
+#include <linux/types.h>
 #include <linux/audit.h>
 
 static const char *errstr;
--- strace-4.18/printsiginfo.c.orig	2017-10-27 11:41:52.332444289 +0200
+++ strace-4.18/printsiginfo.c	2017-10-27 13:10:37.340532514 +0200
@@ -38,6 +38,7 @@
 #include DEF_MPERS_TYPE(siginfo_t)
 
 #include <signal.h>
+#include <linux/types.h>
 #include <linux/audit.h>
 
 #include MPERS_DEFS
--- strace-4.18/tests-m32/netlink_audit.c.orig	2017-10-27 11:42:00.859207485 +0200
+++ strace-4.18/tests-m32/netlink_audit.c	2017-10-27 13:10:37.341532486 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <sys/socket.h>
+#include <linux/types.h>
 #include <linux/audit.h>
 #include "netlink.h"
 
--- strace-4.18/tests-m32/ptrace.c.orig	2017-10-27 11:42:09.485967898 +0200
+++ strace-4.18/tests-m32/ptrace.c	2017-10-27 13:10:37.342532458 +0200
@@ -38,6 +38,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include "ptrace.h"
+#include <linux/types.h>
 #include <linux/audit.h>
 
 static const char *errstr;
--- strace-4.18/tests-mx32/netlink_audit.c.orig	2017-10-27 11:42:17.680740322 +0200
+++ strace-4.18/tests-mx32/netlink_audit.c	2017-10-27 13:10:37.342532458 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <sys/socket.h>
+#include <linux/types.h>
 #include <linux/audit.h>
 #include "netlink.h"
 
--- strace-4.18/tests-mx32/ptrace.c.orig	2017-10-27 11:42:23.797570448 +0200
+++ strace-4.18/tests-mx32/ptrace.c	2017-10-27 13:10:37.343532430 +0200
@@ -38,6 +38,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include "ptrace.h"
+#include <linux/types.h>
 #include <linux/audit.h>
 
 static const char *errstr;
