--- a/printsiginfo.c	2017-10-27 11:41:52.332444289 +0200
+++ b/printsiginfo.c	2017-10-27 13:10:37.340532514 +0200
@@ -38,6 +38,7 @@
 #include DEF_MPERS_TYPE(siginfo_t)
 
 #include <signal.h>
+#include <linux/types.h>
 #include <linux/audit.h>
 
 #include MPERS_DEFS
--- a/netlink.c	2017-11-23 14:53:07.431758917 +0100
+++ b/netlink.c	2017-11-23 14:55:18.353115125 +0100
@@ -28,6 +28,7 @@
 
 #include "defs.h"
 #include <sys/socket.h>
+#include <linux/types.h>
 #include <linux/netlink.h>
 #include "xlat/netlink_flags.h"
 #include "xlat/netlink_types.h"
--- a/tests/ptrace.c	2017-11-23 14:53:11.382648956 +0100
+++ b/tests/ptrace.c	2017-11-23 14:56:13.682575200 +0100
@@ -29,6 +29,7 @@
 
 #include "tests.h"
 #include <asm/unistd.h>
+#include <linux/types.h>
 
 #ifdef __NR_rt_sigprocmask
 
--- a/util.c	2017-11-23 15:01:34.832636976 +0100
+++ b/util.c	2017-11-23 15:10:03.983479256 +0100
@@ -35,6 +35,7 @@
 #include <sys/param.h>
 #include <fcntl.h>
 #include <stdarg.h>
+#include <linux/types.h>
 #ifdef HAVE_SYS_XATTR_H
 # include <sys/xattr.h>
 #endif
@@ -1066,24 +1067,6 @@
 }
 
 static bool process_vm_readv_not_supported = 0;
-#ifndef HAVE_PROCESS_VM_READV
-/*
- * Need to do this since process_vm_readv() is not yet available in libc.
- * When libc is be updated, only "static bool process_vm_readv_not_supported"
- * line should remain.
- */
-/* Have to avoid duplicating with the C library headers. */
-static ssize_t strace_process_vm_readv(pid_t pid,
-		 const struct iovec *lvec,
-		 unsigned long liovcnt,
-		 const struct iovec *rvec,
-		 unsigned long riovcnt,
-		 unsigned long flags)
-{
-	return syscall(__NR_process_vm_readv, (long)pid, lvec, liovcnt, rvec, riovcnt, flags);
-}
-# define process_vm_readv strace_process_vm_readv
-#endif /* !HAVE_PROCESS_VM_READV */
 
 static ssize_t
 vm_read_mem(const pid_t pid, void *const laddr,
